apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/helper/ConnectionMetaData.java

```diff
@@ -208,7 +318,16 @@ ConnectionMetaData parse(String connectionUrl, String user) {
         MYSQL("mysql") {
             @Override
             ConnectionMetaData parse(String connectionUrl, String user) {
-                return ConnectionUrlParser.defaultParse(connectionUrl, dbVendor, 3306, user);
+                String host = "localhost";
+                int port = 3306;
+                HostPort hostPort = parseMySqlFlavor(connectionUrl);
```

Comment:
[minor] same here, having a `null` return value would avoid duplication of the `null` check and `port > 0` below.

Reply:
I don't think-  most (or all) make port configuration optional, each provider using a different default port if not configured.